src/spikeinterface/core/baserecording.py

```diff
@@ -509,6 +509,26 @@ def reset_times(self):
             rs.t_start = None
             rs.sampling_frequency = self.sampling_frequency
 
+    def shift_start_time(self, shift, segment_index=None):
+        """
+        Shift the starting time of the times.
+
+        shift : int | float
+            The shift to apply to the first time point. If positive,
+            the current start time will be increased by `shift`. If
+            negative, the start time will be decreased.
+
+        segment_index : int | None
```

Comment:
Interesting, I was thinking that all segments should be shift at the same times but  adding more fine grain control might make sense. 

How are you thinking about this? Do you think having the default as shifting all the segments makes sense?

Reply:
Good point I wasn't really thinking of this in reference to a real world use-case. My only thinking was if you want to shift all segments, this approach allows you to (a bit verbose though) but vice-versa, you are stuck if the implementation automatically shifts all. I can imagine if you are starting/stopping a recording and storing e.g. for long trial-like periods with the parts in different segments, you may want to adjust each segment separately to align with some trigger?

Reply:
That's a good point. Having control per segment sounds like a good idea. What do you think of having `segment_index=None` shift all the segments by default?

Reply:
Yes that's much nicer, I think that makes more sense than `segmnet_index=None` only shifting the first segment or something. Should `segment_index` be default `None` or a required argument to make the behaviour explicit?

Reply:
My preference is by default you shift everything and then shift a segment if you specify. But I am highly biased because that makes my workflow convenient and I have not think this through. A somewhat weak argument for it is that if I say "shift_times" I want to shift all the times of the object.

Reply:
I think I may lean toward Heberto on this. Why the first segment if `None`. There is no obvious logic for that that I can see? I think the options are None does all or you require explicit. I don't think `None` should implicitly choose a segment for the user. But that's just how I would think to use this as a user.

For a use case maybe I do an `AppendRecording` (I forget the actual class name haha) where I fuse three monosegments together but realized I need to shift them. I like None in this case if I need to move all of them as once otherwise if we don't have a None option then I would have to do a for loop going through each segment. If I only need to do one of the segments I always have the option to specify it rather than have `None` or 0 being the first segment.

Reply:
Great thanks both I agree this feels natural, let's have `None` shift all segments and the default `segment_index=None`.

BTW, a complete aside. I just noticed `times = recording.get_times()` returns a veiw to the time vector (if it exists) otherwise an auto-generated vector of times. Therefore if you edit the `times` (and the recording has a time vector) you'll actually edit the times on the recording. I think this is probably a bad idea, and also means there is different behaviour using `time_vector` vs. `t_start` (in which editing the returned `times` does not change anything on the recording itself). 

What do people think of introducing a `copy=True` (i.e. default on) option for `get_times()` that will make a copy of  `time_vector` (if it exists). If `copy=False`, the time vector will not be copied. Unfortunately this will leak the implementation to the docstring, but I guess this option is necessary in the case of very large time arrays.

Reply:
Mmm interesting point. I need to think on this.  My first instinct is that I would rather not have an argument and always return a copy.

Reply:
I agree actually I think that is much better, much cleaner. There's no point trying to preemptively support a super-edge case (and in such a case the time vector can be accessed directly anyways).

Reply:
Yes, internally for memory efficient reasons we can access the attribute or create a private method. Great point!